plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs

```diff
@@ -191,17 +196,26 @@ instance Pretty EvaluationError where
     pretty (IncompatibleVersionError actual) = "This version of the Plutus Core interface does not support the version indicated by the AST:" <+> pretty actual
     pretty CostModelParameterMismatch = "Cost model parameters were not as we expected"
 
+{-| A script that can be understood by the CEK evaluator.
```

Comment:
How about this:

A variant of `Script` with a specialized `Serialise` instance that decodes the names directly into `NamedDeBruijn`s rather than `DeBruijn`s. This is needed because the CEK machine expects `NameDeBruijn`s, but there are obviously no names in the serialized form of a `Script`. Rather than traverse the term inserting fake names after deserializing, this lets us do at the same time as deserializing.